ambry-store/src/main/java/com.github.ambry.store/Journal.java

```diff
@@ -154,6 +167,23 @@ void addEntry(Offset offset, StoreKey key) {
     return journalEntries;
   }
 
+  /**
+   * @return all journal entries at the time when this method is invoked. (Return empty list if journal is empty)
+   */
+  List<JournalEntry> getAllEntries() {
+    List<JournalEntry> result = new ArrayList<>();
+    // get current last Offset
+    Offset lastOffset = getLastOffset();
+    if (lastOffset != null) {
+      // get portion view of journal whose key is less than or equal to lastOffset
+      NavigableMap<Offset, StoreKey> journalView = journal.headMap(lastOffset, true);
+      for (Map.Entry<Offset, StoreKey> entry : journalView.entrySet()) {
+        result.add(new JournalEntry(entry.getKey(), entry.getValue()));
```

Comment:
I talked offline with Yingyi about possibly passing in a lambda that would test if the journal entry corresponds to a put entry and returning a single JournalEntry instead of a full snapshot. We discussed that this would not work since the ordering of taking the snapshot before searching through validIndexSegments is important in this case.